### ISSUE TITLE
[Error] PDF cover cannot be shown

### ISSUE THREAD

**Issue by seuplus** [geek1011/BookBrowser#35]
Platform:  Debian GNU/Linux 8.10 (jessie)

**1、[error]** PDF cover cannot be shown correctly. See the `Picture1` below.
**2、[enhancement]** Can you add a go-back-to-cover button on epub browser. See the `Picture2` below.

Thank you!

**`Picture1`**
![pdf-cover-error](https://user-images.githubusercontent.com/26323968/34639228-e957bed0-f316-11e7-966e-c8c299150d9e.png)

**`Picture2`**
![screen shot 2018-01-06 at 19 22 40 1 1](https://user-images.githubusercontent.com/26323968/34639257-9c1a198c-f317-11e7-875b-1e5a1c44f880.png)

**Comment by geek1011** [comment]
For the pdf cover, currently there is no way to extract it (there are no libraries written in go) short of writing a whole pdf rendering library (which would take a lot of work) or calling an external command (which will greatly decrease the indexing speed, and will not work on all systems). If I find a PDF library written in go, I will add covers.

For your second issue, it actually belongs in http://github.com/geek1011/ePubViewer/issues . You can view the cover by pressing the info button in the top left, or you can go to it in the table of contents. As for clicking the book title, may I ask why this would be useful?

**Comment by seuplus** [comment]
Thank you!

**Comment by LiMinggang** [comment]
Please make high priority to issue #47 which is a big performance improvement for huge books like mine. I don't think the projects I listed are mature enough for use.
And I think I will try a workaround for the thumbnail issue of PDF, ie, use ghost script to generate thumbnails for BookBrowser. Anyway, the book should not update too often.

**Comment by geek1011** [comment]
Yep. I was thinking of something like that. I'll probably make it call imagemagick if it is present.

As for #47, I'm working on it when I have time. I've figured out how to do an efficient checksum of the epub without reading the whole thing, and now, I just need to figure out how I'm going to store metadata in a forwards compatible way.

**Comment by LiMinggang** [comment]
Thanks for your hard word!

**Comment by LiMinggang** [comment]
Another candidate in go
https://github.com/hhrutter/pdfcpu

**Comment by geek1011** [comment]
@LiMinggang That looks almost perfect! I'll see how it works later today.

**Comment by LiMinggang** [comment]
There is another JS version which supports PDF thumbnail. No idea if it's OK for us.
https://github.com/oguzhaninan/Buka

**Comment by geek1011** [comment]
No, that won't work. It uses PDF.js to render it. Last time I tried, I couldn't get it running in otto (a js runtime for go).
